src/cljs/org/broadinstitute/firecloud_ui/main.cljs

```diff
@@ -126,24 +135,24 @@
              [:div {}
               [TopNavBar {:selected-item page
                           :on-nav (fn [item] (nav/navigate (:nav-context props) (name item)))}]
-              (let [item (first (filter #(= (% :key) page) top-nav-bar-items))]
+              (let [item (first (filter #(= (% :key) page) routes))]
                 (if item
                   ((item :render) {:nav-context nav-context})
                   [:div {} "Page not found."]))]))]]))
    :component-did-mount
    (fn [{:keys [this state]}]
      (endpoints/profile-get
-      :isRegistrationComplete
-      (fn [{:keys [success? status-text get-parsed-response]}]
-        (cond
-          (and success? (= (get-in (get-parsed-response) ["keyValuePair" "value"]) "true"))
-          (swap! state assoc :registration-status :registered)
-          (not success?)
-          (swap! state assoc :registration-status :not-registered)
-          :else
-          (do
-            (set! (.-errorMessage this) status-text)
-            (swap! state assoc :registration-status :error))))))})
+       (fn [{:keys [success? status-text get-parsed-response]}]
+         (let [parsed-values (common/parse-profile (get-parsed-response))]
+           (cond
+             (and success? (= (:isRegistrationComplete parsed-values) "true"))
+             (swap! state assoc :registration-status :registered :name (:name parsed-values))
+             (not success?)
+             (swap! state assoc :registration-status :not-registered)
```

Comment:
@dmohs - I'm not following this suggestion. The way I read the code, this is what happens:

* registered key exists -> assoc registered and name values - move user along to requested page (lines 132-140)
* registered key does not exist -> assoc not-registered and let the component render the registration page (lines 128, 129)
* Anything else (i.e. a server error when making the get-profile call) puts the error into :registration-status, which is then handled in lines 126 and 127.

As I see it, a server error should trigger line 127.